test/apps/app_output_verifier.cpp

```diff
@@ -0,0 +1,137 @@
+#include "teca_config.h"
+#include "teca_cf_reader.h"
+#include "teca_cartesian_mesh_reader.h"
+#include "teca_cartesian_mesh_writer.h"
+#include "teca_table_reader.h"
+#include "teca_table_writer.h"
+#include "teca_table_sort.h"
+#include "teca_dataset_diff.h"
+#include "teca_file_util.h"
+#include "teca_mpi_manager.h"
+#include "teca_system_interface.h"
+#include "teca_mpi.h"
+
+#include <vector>
+#include <string>
+#include <iostream>
+using namespace std;
+
+
+int main(int argc, char **argv)
+{
+    teca_mpi_manager mpi_man(argc, argv);
+    int rank = mpi_man.get_comm_rank();
+
+    teca_system_interface::set_stack_trace_on_error();
+    teca_system_interface::set_stack_trace_on_mpi_error();
+
+    if (argc < 4)
+    {
+        if (rank == 0)
+        {
+            cerr << endl
+                << "Usage:" << endl
+                << "app_test_verifier [output file type] [app output file] "
+                   "[reference file] [sort by column]" << endl
+                 << endl;
+        }
+        return -1;
+    }
+
+    // parse command line
+    unsigned int out_file_type = atoi(argv[1]);
+    std::string app_out_file = argv[2];
+    std::string ref_file = argv[3];
+    std::string column_name;
+    if (argc == 5) column_name = argv[4];
+    
+
+    if (!teca_file_util::file_exists(app_out_file.c_str()))
+    {
+        cerr << "App Verifier Error: app output file doesn't exist" << endl;
```

Comment:
please use TECA_ERROR for error messages. applies here and elsewhere cerr is used to print error messages